lib/streamlit/elements/button.py

```diff
@@ -39,31 +39,31 @@ class ButtonMixin:
     def button(
         self,
         label,
-        key: Optional[str] = None,
-        help: Optional[str] = None,
-        on_change: Optional[WidgetCallback] = None,
-        args: Optional[WidgetArgs] = None,
-        kwargs: Optional[WidgetKwargs] = None,
+        key=None,
+        help=None,
+        on_change=None,
+        args=None,
+        kwargs=None,
     ) -> bool:
         """Display a button widget.
 
         Parameters
         ----------
         label : str
             A short label explaining to the user what this button is for.
-        key : Optional[str]
+        key : str
```

Comment:
Should we preserve the *Optional* part here?

Reply:
I intentionally removed it to stay consistent with the rest of our documentation, where we don't type our optional fields so accurately and instead just mention that the parameter is optional in its description.

I definitely think there's some merit to making the types here more correct, but it's something that'd probably be best done outside of the session state feature branch since it'll require us to change docstrings for most of our API.